src/internal.c

```diff
@@ -15651,8 +15651,9 @@ int ProcessReply(WOLFSSL* ssl)
 
             ssl->options.processReply = doProcessInit;
 
-            /* input exhausted? */
-            if (ssl->buffers.inputBuffer.idx >= ssl->buffers.inputBuffer.length)
+            /* input exhausted or app data received */
+            if (ssl->buffers.inputBuffer.idx >= ssl->buffers.inputBuffer.length
+                    || ssl->curRL.type == application_data)
```

Comment:
Since this is for DTLS to return multiple application records in a single datagram separately, you should check for DTLS mode.